### PR TITLE
[Compression] fix tests & bugs

### PR DESCRIPTION
**Pull request by J-shang**
#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

File: nni/common/concrete_trace_utils/concrete_tracer.py [hunk 1]
```diff
@@ -1371,6 +1380,18 @@ def f(x, y):
...
+    args = inspect.getfullargspec(root.forward).args[1:]
+    defaults = inspect.getfullargspec(root.forward).defaults
+    if isinstance(concrete_args, (tuple, list)):
+        concrete_args = (*concrete_args, *defaults[len(concrete_args) + len(defaults) - len(args):])
+    else:
+        kv_default = {k: v for k, v in zip(args[-len(defaults):], defaults)}
+        concrete_args = {
+            **concrete_args,
+            **{n: kv_default[n] for n in args if n not in concrete_args}
+        }
```
Comment (super-dainiu) [inline nni/common/concrete_trace_utils/concrete_tracer.py:1393]: better put this inside ``ConcreteTracer().trace()``?

Comment (J-shang) [reply nni/common/concrete_trace_utils/concrete_tracer.py:1393]: good suggestion

File: nni/common/concrete_trace_utils/operator_patcher.py [hunk 2]
```diff
@@ -226,20 +229,27 @@ def patch_inner_helper(self, func):
...
+                        **closure_dict,
+                    },
+                    var_dict)
+                if the_self is not None:
+                    return var_dict['new_func'].__get__(the_self)
+                else:
+                    return var_dict['new_func']
+            finally:
+                if platform.python_version_tuple() < ('3', '9'):
+                    setattr(builtins, 'tuple', tuple_wrapped)
```
Comment (super-dainiu) [inline nni/common/concrete_trace_utils/operator_patcher.py:252]: what are these for?

Comment (J-shang) [reply nni/common/concrete_trace_utils/operator_patcher.py:252]: the type hint `typing.Tuple` will meet error if the tuple is wrapped under `python < 3.9`

**Review by Bonytu — APPROVED** [review]

**Review by ultmaster — APPROVED** [review]

**Pull request merged**